### ISSUE TITLE
Problem about GPU memory

### ISSUE THREAD

**Issue by Ruikun-Li** [milesial/Pytorch-UNet#52]
Hi milesial! Thanks for your great works. I learned much with it.
When I want to train with my own data, i found the image size is really huge (4288*2848) so i resize the image as you mentioned in README. But when I used bilinear upsample instead of transposed convolution, the GPU memory using didn`t change at all（about 5GB）. Do u have any suggestions about it?
Thank you.

**Comment by milesial** [comment]
Hi, if you use `bilinear=True` as it is by default, the memory usage should be as low as it can get. I don't know why that doesn't change for you.
All I can suggest is lowering the batch_size (even 1 is an option) and lowering again the size of your images if memory is a problem.

**Comment by Ruikun-Li** [comment]
Thank you for your suggestions. Lowering the batch_size didn`t seems work.
I tried to analyze the number of parameters. There were 14788929 params when i used transposed convolution and 13395329 params with bilinear upsample. The number decreased about 10%, and the size of the model i saved also decreased about 10% from 57826KB to 52381KB. In this situation, my GPU memory usage decreased about 8 MB from 1888MB to 1880MB. During several times attempt, the results are same.
So I suppose maybe using bilinear upsample won`t bring huge promotion about the memory use(just about 10%)? But i am still confused with the GPU memory usage because it just decreased 8MB.
After all, thank you for your codes and help. If i figure it out later, i will update this issue.

**Comment by Ruikun-Li** [comment]
I think I found an explanation.

The number of the parameters is about 13M~15M, so the memory usage is just about 14M*4/1024/1024=53M, which means the GPU memory usage caused by parameters is really low. The major usage should be the forward and backward pass size.

I check the memory usage with [pytorchsummary,](https://github.com/sksq96/pytorch-summary) the result confirms my assumption.

Using bilinear upsample:
```
Total params: 13,395,329
Trainable params: 13,395,329
Non-trainable params: 0
Input size (MB): 1.39
Forward/backward pass size (MB): 1724.44
Params size (MB): 51.10
Estimated Total Size (MB): 1776.93
```

Using transposed convolution:
```
Total params: 14,788,929
Trainable params: 14,788,929
Non-trainable params: 0
Input size (MB): 1.39
Forward/backward pass size (MB): 1724.44
Params size (MB): 56.42
Estimated Total Size (MB): 1782.25
```

The parameters size decreased exactly about 10%, consistent with previous analysis.
Another possible conclusion: With these configurations, replacing transposed convolution with bilinear upsample won`t influence the GPU memory usage to much.

**Comment by milesial** [comment]
Yes that's a nice summary there ! So it seems the fastest way to decrease memory is to scale down the images and masks.

**Comment by Ruikun-Li** [comment]
Completely agree! :)
